File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int start=0, maxCount =0, count=0,len=0;
        for(int end =0; end< nums.length; end++){
            if(nums[end] == 1)
                count++;
            maxCount = Math.max(maxCount, count);
            while(end- start+1 -maxCount> k){
                if(nums[start]==1)
                    count--;
                start++;
            }
            len = Math.max(len, end-start+1);
        }
        return len;
    }
}

File: 1004-max-consecutive-ones-iii/1004-max-consecutive-ones-iii.py
class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        maxlen = 0
        numOfZeroes = 0
        start = 0

        for end in range(len(nums)):
            if nums[end] == 0:
                numOfZeroes += 1
            
            while numOfZeroes - k > 0: # sliding window, if number of zeroes goes above k 
                if nums[start] == 0:
                    numOfZeroes -= 1
                start += 1
            
            maxlen = max(maxlen, end - start + 1)
        
        return maxlen
